crates/mod-crypto/src/lib.rs

```diff
@@ -0,0 +1,375 @@
+use anyhow::anyhow;
+use aws_lc_rs::cipher::{
+    AES_128, AES_256, DecryptionContext, PaddedBlockDecryptingKey, PaddedBlockEncryptingKey,
+    UnboundCipherKey,
+};
+
+use data_encoding::{
+    BASE32, BASE32HEX, BASE32HEX_NOPAD, BASE32_NOPAD, BASE64, BASE64URL, BASE64URL_NOPAD,
+    BASE64_NOPAD, HEXLOWER,
+};
+use config::{any_err, from_lua_value, get_or_create_sub_module};
+use mlua::prelude::LuaUserData;
+use mlua::{Lua, MetaMethod, UserDataFields, UserDataMethods, Value, Error as LuaError};
+use data_loader::KeySource;
+use hex::decode;
+use std::str;
+
+use serde::Deserialize;
+
+#[derive(Deserialize, Clone, Debug)]
+pub struct AesParams {
+    pub algorithm: AesAlgo,
+    pub key: AesKey,
+}
+
+#[derive(Deserialize, Clone, Debug)]
+pub enum AesAlgo {
+    Ecb(),
+    Cbc(),
+}
+
+#[derive(Deserialize, Clone, Debug)]
+pub struct LuaCfg {
+    pub key: KeySource,
+    pub value: String,
+    pub algorithm: String,
+}
+
+#[derive(Deserialize, Clone, Debug)]
+pub struct LuaCfgDecrypt {
+    pub key: KeySource,
+    pub value: Vec<u8>,
+    pub algorithm: String,
+}
+
+#[derive(Deserialize, Clone, Debug)]
+pub enum AesKey {
+    Aes128([u8; 16]),
+    Aes256([u8; 32]),
+}
+
+fn is_hex(vec: &[u8]) -> bool {
+    let hex_str = String::from_utf8_lossy(vec);
+    hex_str.chars().all(|c| c.is_digit(16)) && hex_str.len() % 2 == 0
+}
+
+impl AesKey {
+    pub fn from_bytes(bytes: &[u8]) -> Result<Self, String> {
+        let decoded_bytes = if is_hex(bytes) {
+            decode(bytes).map_err(|e| format!("Failed to decode hex: {}", e))?
+        } else {
+            bytes.to_vec()
+        };
```

Comment:
this sort of thing results in ambiguity; what if I have data that looks like hex (eg: the word `deadbeef`) but which is not hex?  This logic will result in it being treated as hex and interpreted incorrectly.

Can we remove support for implicitly hex encoded keys? That will make things simpler to reason about here in the code.

If you have and need to support hex encoded keys, can we introduce a way to explicitly indicate that the data is hex encoded into the API/parameters?